src/assign.rs

```diff
@@ -58,25 +54,456 @@ pub struct Assignment<'a> {
     ///
     /// Note: if a portion of the path contains a leaf node that is to be
     /// overridden by an object or an array, then the path will be leaf that is
-    /// replaced.
-    ///
-    /// For example, if you had the json:
-    /// ```json
-    /// { "foo:" "bar" }
-    /// ```
-    /// and you assigned `"new_value"` to `"/foo/bar/baz"`, then the value would
-    /// be `Some("/foo/bar")`.
-    pub created_or_mutated: PointerBuf,
-    /// A `Pointer` consisting of the path which was assigned.
+    /// replaced. For example, given the json `{ "foo:" "bar" }`, if `"new_value"` is
+    /// assigned to `"/foo/bar/baz"`, then `assigned_to` would be `/foo/bar"`
     ///
     /// ## Example
     /// ```rust
-    /// use serde_json::json;
-    /// use jsonptr::{Pointer, Assign};
+    /// # use serde_json::json;
+    /// # use jsonptr::{Pointer, Assign};
     /// let mut data = json!({ "foo": ["zero"] });
     /// let mut ptr = Pointer::from_static("/foo/-");
     /// let assignment = data.assign(&mut ptr, "one").unwrap();
     /// assert_eq!(assignment.assigned_to, Pointer::from_static("/foo/1"));
     /// ```
     pub assigned_to: PointerBuf,
+
+    /// The value that was replaced, if any.
+    pub replaced: Option<Value>,
+}
+
+pub(crate) fn assign_value<'v>(
+    mut ptr: &Pointer,
+    mut value: &'v mut Value,
+    mut src: Value,
+) -> Result<Assignment<'v>, AssignError> {
+    let mut offset = 0;
+    let mut buf = PointerBuf::with_capacity(ptr.as_str().len());
```

Comment:
Oh, thanks for mentioning that. I had an error (using `str::len` rather than `Chars::len`).

This does bring up a question as to what should `len` report. `Token` count?, byte count like `str`, or `Chars::len`